bin/remote-prover/src/client/mod.rs

```diff
@@ -0,0 +1,30 @@
+//! Remote prover client implementations
+//!
+//! Provides gRPC clients for interacting with remote proving services.
+
+// Conditional exports based on enabled features
+#[cfg(feature = "tx-prover")]
+pub mod tx_prover;
+#[cfg(feature = "tx-prover")]
+pub use tx_prover::RemoteTransactionProver;
+
+#[cfg(feature = "batch-prover")]
+pub mod batch_prover;
+#[cfg(feature = "batch-prover")]
+pub use batch_prover::RemoteBatchProver;
+
+#[cfg(feature = "block-prover")]
+pub mod block_prover;
+#[cfg(feature = "block-prover")]
+pub use block_prover::RemoteBlockProver;
```

Comment:
Are there additional items within each that are required? At least the `batch_prover` module only contains the `RemoteBatchProver` so we should be able to simplify the exports.